mosaicml/llm-foundry
# add `.json` to SUPPORTED_EXTENSIONS

**Summary:**
This PR allows us to create a finetuning dataset from `.json` files.

**What Happened:**
I was trying to finetune `llama2-7b-chat` on the [OpenHermes-2.5](https://huggingface.co/datasets/teknium/OpenHermes-2.5) dataset but got an error because the dataset is a `.json` file, not a `.jsonl` file.

Looking into it, I discovered that when `is_safe=True`, we check that the dataset files all have extensions in `SUPPORTED_EXTENSIONS` ([source](https://github.com/mosaicml/llm-foundry/blob/f01f6256ba487ef9e433f49ceac68cf327ac74bb/llmfoundry/data/finetuning/tasks.py#L752)). Currently,
```python
SUPPORTED_EXTENSIONS = ['.csv', '.jsonl', '.parquet']
```
and because the OpenHermes-2.5 dataset consists of a `.json` file, which is not in `SUPPORTED_EXTENSIONS`, we get an error.

**My Solution:**
So I added `.json` to the SUPPORTED_EXTENSIONS and now I can finetune on this. This PR is really pretty simple. But I tested this on some runs to make sure everything works.

**Testing:**
My runs (below) show that when you currently finetune on a `.json` files it works when `is_safe=False` (run 2) but fails when `is_safe=True` (run 1). With my changes in the `eitan-patch-json` branch, you can finetune on a `.json` file when `is_safe=True` (run 3).

Runs:
1. Run `llama2-7b-chat-open-hermes-ft-DKoAb6` on `main` branch -- I successfully finetuned on the `.json` in OpenHermes-2.5 when `is_safe=False`.
2. Run `llama2-7b-chat-open-hermes-ft-CY6yNJ` on `main` branch -- I got an error finetuning on the `.json` file in OpenHermes-2.5 when`is_safe=True`.
3. Run `llama2-7b-chat-open-hermes-ft-5xoXNv` on `eitan-patch-json` branch -- I successfully finetuned on the `.json` in  OpenHermes-2.5 when `is_safe=True`  by running on my branch `eitan-patch-json`.

Note: I stopped runs 1. and 3. after training for 10+ batches so the error you see in the logs is from me stopping the run early, not from preparing the dataset. To view the logs of run 2. do `mcli logs llama2-7b-chat-open-hermes-ft-CY6yNJ --resumption 0`. The error from this run looks like:
```bash
InvalidFileExtensionError: safe_load is set to True. No data files with safe 
extensions ['.csv', '.jsonl', '.parquet'] found for dataset at local path 
teknium/OpenHermes-2.5.
```
